### DISCUSSION TITLE
The AI keeps ignoring the 'ignore' command.

### DISCUSSION THREAD

**Discussion by KougamiYuriko** [FakeFishGames/Barotrauma#15057] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When a crew member is injured, the AI doctors ignore the 'ignore' command and aggressively use medical supplies to treat them, resulting in a constant shortage of supplies on the ship.
 
I'm playing on Hell difficulty and don’t have much money to buy medical supplies, so I only use them when health is in actual danger. Even without issuing a treatment command to the AI doctors, they keep consuming my supplies, leaving me without any in critical moments. The 'ignore' command doesn't stop this from happening. The only solution seems to be dismissing or sacrificing the AI doctors.

I've also noticed that the AI doctors misjudge the value of medical supplies; they use fentanyl when health is around 90% and apply plastiseal for minor burns.

"I believe this is a serious AI flaw, and the doctor's AI needs to be improved as soon as possible; otherwise, medical supplies will continue to be wasted by the AI indiscriminately.

### Reproduction steps

1.When a crew member is injured
2.The AI doctors automatically ignore the "ignore" command and take medical supplies

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
What exactly you ordered them to ignore? Medical cabinet?

**Reply by KougamiYuriko** [reply]
I set certain high-value medical supplies to 'ignore' individually, such as fentanyl and Pomegrenade Extract. 
When I didn’t deal with the Thalamus, many crew members were burned, and in one round, the AI doctors used up 36 bandages and 12 plastiseals. I hadn’t issued any treatment commands to the AI doctors, and I also set bandages and those items as ignored targets, yet they still used up these supplies.

Please excuse the fact that I’m using the Chinese version and a cosmetic mod, but these don’t affect the outcome.
![20241105022503_1](https://github.com/user-attachments/assets/9aef507d-b3a1-4f4c-afa2-cc18216ef802)

**Comment by SomeRandomNoobKekeke** [comment]
I can't reproduce, they seems to ignore medical supplies and let their patients die as intended
With ignore order on individual items or on the container
![20241104213526_2](https://github.com/user-attachments/assets/afcd4ddb-bc15-43d6-ae32-28052c8ee5d4)
There is a nuance: Ignore order, in fact, is applied to individual items
When you order to ignore a stack - ignore order is applied to whole stack
But if you add more items to the stack they will be placed behind ignored items, it may look like the whole stack is ignored, but those normal items don't have individual orders and they won't be ignored

You need to give another order every time, or you can order to ignore, for example, toxin cabinet and everything inside will be ignored

**Reply by KougamiYuriko** [reply]
Wow, so that's how it works—each item has its own ignore command. I always thought the command applied to item stacks. The system didn't give any hints about this command.
Is there a chance to improve the UI in the future? For example, when new items stack with ignored items, the icon could turn yellow or something similar.
Thank you for answering this question.

**Reply by SomeRandomNoobKekeke** [reply]
Yes, it's very confusing, it also works the same way with stolen items

> Is there a chance to improve the UI in the future?

Need to ask the actual devs @JeroonBosch

**Comment by pavel4win** [comment]
There is actually a bug here.
It ONLY manifests itself in the NETWORK GAME!
In a single player game, everything works as it should.
I already created a discussion about it

https://github.com/FakeFishGames/Barotrauma/discussions/15074

But I just tested and reveal, this bug is not fixed in the latest version.
So, in a network game, to reproduce bug:
Put a stack of items in the closet.
Give the command "ignore" to this stack.
Then drop it on the floor.
Tell the bot to "clean up"
You will see that the bot will take this stack and put it back in the closet.
That is, the command "ignore" did not work on the stack.
For the purity of the experiment, take another stack.
Now scatter this stack one by one.
You can scatter it in a cabinet, or in your inventory, no matter.
Give the command "ignore" to each item.
Now combine them into a stack.
Drop this stack on the floor.
Tell the bot to "clean up"
The bot will ignore this stack as expected.
This bug hasn't been fixed yet.
The only way to guarantee "ignore" items in network game
Is to give the "ignore" command to every item in the stack.
A simpler method is to deny bots access to a certain cabinet
and keep the meds in that cabinet.
This method works as expected.
And one more piece of advice.
Give each bot a stack of pomegranate extract.
And give the doctor 3 stacks.
If he has the right medicine in his inventory, he won't run to the first aid kit.

**Comment by Regalis11** [comment]
Thank you for the report! This seems to be a symptom of the same bug reported in https://github.com/FakeFishGames/Barotrauma/discussions/14427: marking a stack of items to be ignored actually only marks the first item in the stack. We've got a fix implemented for this in our private repository, but it unfortunately isn't available in the public build yet. Sorry for the inconvenience!
